exercises/concept/library-fees/.meta/exemplar.ex

```diff
@@ -0,0 +1,41 @@
+defmodule LibraryFees do
+  def datetime_from_string(string) do
+    NaiveDateTime.from_iso8601!(string)
+  end
+
+  def before_noon?(datetime) do
+    Time.compare(NaiveDateTime.to_time(datetime), ~T[12:00:00]) == :lt
+  end
+
+  def return_datetime(checkout_datetime) do
+    days = if before_noon?(checkout_datetime), do: 28, else: 29
+    NaiveDateTime.add(checkout_datetime, days * 24 * 60 * 60, :second)
+  end
+
+  def days_late(planned_return_datetime, actual_return_datetime) do
+    diff =
+      Date.diff(
+        NaiveDateTime.to_date(actual_return_datetime),
+        NaiveDateTime.to_date(planned_return_datetime)
+      )
+
+    if diff < 0, do: 0, else: diff
+  end
+
+  def monday?(datetime) do
+    datetime
+    |> NaiveDateTime.to_date()
+    |> Date.day_of_week()
+    |> Kernel.==(1)
+  end
+
+  def calculate_late_fee(checkout, return, rate) do
+    checkout_datetime = datetime_from_string(checkout)
+    planned_return_datetime = return_datetime(checkout_datetime)
+    actual_return_datetime = datetime_from_string(return)
+
+    days_late = days_late(planned_return_datetime, actual_return_datetime)
+    rate = if monday?(actual_return_datetime), do: 0.5 * rate, else: rate
+    trunc(days_late * rate)
+  end
+end
```

Comment:
I was wondering if this is maybe a bit too much? I could easily remove either the Monday condition or the before noon condition to make the exercise a bit simpler. I included the Monday condition to teach about `Date.day_of_week()` and the noon condition to force the student to write a time literal with the `~T` sigil and to do anything related to time.

Reply:
I don't think it is too much from the perspective of any one function, but it is a longer exercise overall than some of our other ones. Maybe that is what you are sensing?

Reply:
Possibly. After I thought about it again, the exercise feels simple enough, I think there aren't any tricky steps.